e2e/test/scenarios/onboarding/navbar/whats-new.cy.spec.js

```diff
@@ -0,0 +1,84 @@
+import { navigationSidebar, restore } from "e2e/support/helpers";
+import {
+  createMockVersionInfo,
+  createMockVersionInfoRecord as mockVersion,
+} from "metabase-types/api/mocks";
+
+describe("nav > what's new notification", () => {
+  beforeEach(() => {
+    restore();
+
+    mockVersions({
+      currentVersion: "v0.48.0",
+      versions: [
+        mockVersion({
+          version: "v0.48.0",
+          announcement_url: "metabase.com/releases/48",
+        }),
+        mockVersion({
+          version: "v0.47.1",
+          announcement_url: "metabase.com/releases/47",
+        }),
+      ],
+    });
+  });
+
+  // replace `skip` with `only` to manually test this feature with the mocks set up
+  it.skip("mock setup for manual tests", () => {
+    cy.signInAsAdmin();
+
+    loadHomepage();
+
+    navigationSidebar().findByText("See what's new");
+
+    cy.pause();
+  });
+
+  it("should show a notification with a link to the release notes, and allow the dismissal of it", () => {
+    cy.signInAsAdmin();
+
+    loadHomepage();
+    navigationSidebar().findByText("See what's new");
+
+    // should persist reloads
+    loadHomepage();
+    navigationSidebar().findByText("See what's new");
+
+    navigationSidebar().icon("close").click();
+    navigationSidebar().findByText("See what's new").should("not.exist");
+
+    loadHomepage();
+    navigationSidebar().findByText("See what's new").should("not.exist");
+  });
+
+  it("it should show the notification for other users after one user dismissed it", () => {
+    cy.signInAsAdmin();
+    loadHomepage();
+    navigationSidebar().findByText("See what's new");
+    navigationSidebar().icon("close").click();
+
+    cy.signInAsNormalUser();
+    loadHomepage();
+    navigationSidebar().findByText("See what's new");
+  });
+});
+
+function mockVersions({ currentVersion, versions = [] }) {
+  const [latest, ...older] = versions;
+  cy.intercept("GET", "/api/session/properties", req => {
+    req.reply(res => {
+      res.body["version"] = { tag: currentVersion };
+      res.body["version-info"] = createMockVersionInfo({ latest, older });
+    });
+  }).as("sessionProperties");
+}
+
+function loadHomepage() {
+  cy.visit("/");
+
+  // make sure page is loaded
+  cy.findByText("loading").should("not.exist");
```

Comment:
These should probably go in reverse order - with `wait` first. The difference is that `cy.wait` waits for much longer is the request was made. We have many flaky tests because of regular `findByText` checks timeouts